crud/select.lua

```diff
@@ -137,32 +146,36 @@ local function build_select_iterator(space_name, user_conditions, opts)
     if space == nil then
         return nil, SelectError:new("Space %s doesn't exists", space_name)
     end
-
     local space_format = space:format()
 
-    local after_tuple = utils.flatten(opts.after, space_format)
-
     -- plan select
-    local plan, err = select_plan.new(space, conditions, {
-        limit = opts.limit,
-        after_tuple = after_tuple
-    })
+    local plan, err = select_plan.gen_by_conditions(space, conditions)
 
     if err ~= nil then
         return nil, SelectError:new("Failed to plan select: %s", err)
     end
 
-    -- get replicasets to select from
-    local replicasets, err = get_replicasets_to_select_from(plan, replicasets)
-    if err ~= nil then
-        return nil, SelectError:new("Failed to get replicasets to select from: %s", err)
-    end
+    -- set limit and replicasets to select from
+    local total_tuples_count = opts.limit
+    local replicasets_to_select = replicasets
 
-    local scan_index = space.index[plan.scanner.index_id]
+    local scan_index = space.index[plan.index_id]
     local primary_index = space.index[0]
+    if select_plan.is_scan_by_full_sharding_key_eq(plan, scan_index, primary_index) then
+        total_tuples_count = 1
+        plan.iter = box.index.REQ
+
+        local sharding_key_value = utils.extract_subkey(
```

Comment:
Simply `sharding_key`

You don't use `parts_value`, `scan_value_value` or `space_value` in any places